src/js/state/migrations/202005271240_setHistorySpaceToNameAndId.test.js

```diff
@@ -0,0 +1,31 @@
+/* @flow */
+
+import getTestState from "../../test/helpers/getTestState"
+import migrate from "./202005271240_setHistorySpaceToNameAndId"
+import {cloneDeep} from "lodash"
+
+const getTabHistoryEntries = (state) => {
+  return state.tabs.data.map((t) => t.history.entries)
+}
+
+test("migrating 202005271240_setHistorySpaceToNameAndId", () => {
+  const prev = getTestState("v0.9.1")
+  const next = migrate(cloneDeep(prev))
```

Comment:
@jameskerr side note, how do you feel about making `migrate()` more "pure" by always returning a new copy? I personally prefer that sort of functional paradigm which, as an example, would make our tests more idiomatic since we often will compare prev to next.

Reply:
In general, I like as little mutation as possible. However, I'd like to be able to mutate the object in the migration file. I think (in javascript), mutative code is easier to read and understand. So if we were to implement that, I think doing something like you're doing here (cloning it right before you pass it to the migrate function) is ideal. 

Related to the need to use deepClone, I'll share something I've heard about writing tests. They said it's helpful if tests don't repeat the implementation of the code they test. Instead, they assert against hardcoded, expected values. This makes them more likely to catch errors, and easier to read, and won't change if the undelying code changes. There may have been other good reason's but I can't think of them.

FWIW, your test here is totally fine with me, but I'll throw out this alternative.

```js
  const prev = getTestState("v0.9.1")
  const next = migrate(prev)

  let windows = Object.values(next.windows)
  let tab1 = getTabHistoryEntries(windows[0].state)[0]
  let tab2 = getTabHistoryEntries(windows[0].state)[1]

  expect(tab1.every((e) => e.spaceId === "corelight.pcap.brim")).toBe(true)
  expect(tab1.every((e) => e.spaceName === "corelight.pcap.brim")).toBe(true)

  expect(tab2.every((e) => e.spaceId === "pcaps.brim")).toBe(true)
  expect(tab2.every((e) => e.spaceName === "pcaps.brim")).toBe(true)
```

I see that it's a little more difficult here because there's an array of windows and for each of those and array of tabs to test. Anyway, in the case of testing against hardcoded values, you wouldn't need to use the previous value in the assertions.

Anyway, this PR looks good. I just wanted to share that thought.